pipeline_utils/pipeline_deploy.py

```diff
@@ -0,0 +1,414 @@
+                     version='VERSION', pipeline='PIPELINE', kms_key_id=None):
+    """
+        post | patch metadata and docker from a pipeline repo
+            to the specified environment (ff_key)
+
+        the repo must follow the structure described in the header
+    """
+
+    # Get pipeline version
+    with open(repo + '/' + version) as f:
+        version = f.readlines()[0].strip()
+    # Get pipeline name
+    with open(repo + '/' + pipeline) as f:
+        pipeline = f.readlines()[0].strip()
+
+    # Software
+    if post_software:
+        _post_patch_software(ff_key, repo, project,
+                             institution, verbose, debug)
+
+    # File format
+    if post_file_format:
+        _post_patch_file_format(ff_key, repo, project,
+                                institution, verbose, debug)
+
+    # File reference
+    if post_file_reference:
+        _post_patch_file_reference(ff_key, repo, project,
+                                   institution, verbose, debug)
+
+    # Workflow
+    if post_workflow:
+        _post_patch_workflow(ff_key, repo, project, institution,
+                             version, pipeline, wfl_bucket, verbose, debug)
+
+    # Metaworkflow
+    if post_metaworkflow:
+        _post_patch_metaworkflow(ff_key, repo, project, institution,
+                                 version, verbose, debug)
+
+    # Wfl
+    if post_wfl:
+        _post_patch_wfl(version, repo, pipeline, account,
+                        region, wfl_bucket, sentieon_server, kms_key_id=kms_key_id)
+
+    # ECR
+    if post_ecr:
+        _post_patch_ecr(version, repo, account, region)
+
+
+def main(args):
+    """
+        deploy cgap pipeline
+        post | patch metadata and dockers in the specified environment from repos
+    """
+    # # Get credentials
+    # if os.environ.get('GLOBAL_BUCKET_ENV') and os.environ.get('S3_ENCRYPT_KEY'):  # new cgap account
+    #     s3 = s3_utils.s3Utils(env=args.ff_env)
+    #     ff_key = s3.get_access_keys('access_key_admin')
+    # elif os.environ.get('S3_ENCRYPT_KEY'):  # main account, also need the key to get auth this way
+    #     ff_key = ff_utils.get_authentication_with_server(ff_env=args.ff_env)
+    # else:
+    #     # read key for portal auth from args.keydicts_json [~/.cgap-keydicts.json]
+    #     try:
+    #         with open(os.path.expanduser(args.keydicts_json)) as keyfile:
+    #             keys = json.load(keyfile)
+    #     except Exception:
+    #         error = 'ERROR, could not locate file with key dicts for portal auth --keydicts-json\n'
+    #         sys.exit(error)
+    #     ff_key = keys.get(args.ff_env)
+    #
+    # # Get encryption key
+    # kms_key_id = os.environ.get('S3_ENCRYPT_KEY_ID', None)
+    # # Check args
+    # if not ff_key:
+    #     error = 'ERROR, missing key for {0} environment in key dicts for auth\n'.format(args.ff_env)
+    #     sys.exit(error)
+
+    kms_key_id, ff_key = '', {}
```

Comment:
How does this code work without this?

Reply:
it does not, that will be turned on again! I just turned it off for now to not bother with credentials as I am only printing the jsons for testing and check everything is working fine before the actual posting